### DISCUSSION TITLE
Is there an API to retry/resubmitting manually in addition to application startup?

### DISCUSSION THREAD

**Discussion by karmingc** [spring-projects/spring-modulith#207] [Q&A]
According to the API docs, it seems to be possible to query the incomplete publications with [DefaultEventPublicationRegistry's findIncompletePublications()](https://docs.spring.io/spring-modulith/docs/current/api/org/springframework/modulith/events/DefaultEventPublicationRegistry.html). However,  [By default, all incomplete event publications are resubmitted at application startup](https://docs.spring.io/spring-modulith/docs/0.1.0-M1/reference/html/#events.publication-registry).

I can't seem to find more information about triggering it manually or resubmitting incomplete publications and would appreciate any pointers.

**Comment by MaksimMyshkin** [comment]
The only code I found is that triggering by spring.modulith.republish-outstanding-events-on-restart property. It placed in org.springframework.modulith.events.support.PersistentApplicationEventMulticaster.afterSingletonsInstantiated()/invokeTargetListener() but second method is private and it is no way to use it.

I think it would be nice if this functionality was public. In current state the only option that I see is to copy parts of library code to recreate listeners restart functionality.

**Comment by matiwinnetou** [comment]
I also found it strange in a way, re-queueing failed events could be via some exponential back-off enabled strategy triggered periodically or at least expose interface in the application code to do this.
